build-aux/inline-source

```diff
@@ -118,27 +119,38 @@ func_include ()
     _G_scriptdir=`echo "$1" |$SED 's|[^/]*$||'`
     test -n "$_G_scriptdir" || _G_scriptdir="./"
 
+    # quote progpath for use in AWK script
+    func_quote_arg unquoted "$progpath"
+    _G_qprogpath=$func_quote_arg_unquoted_result
+
     $AWK '
-        BEGIN { magic = '${_RECURSE_MAGIC-0}'; }
+        BEGIN { magic = '"${_RECURSE_MAGIC-0}"'; }
 
         /^#!/ && magic == 0 {
             print $0;
-            print "## DO NOT EDIT - This file generated from '$1'";
-            print "##               by '$progname' v'$scriptversion'";
+            print "## DO NOT EDIT - This file generated from '"$1"'";
+            print "##               by '"$progname"' v'"$scriptversion"'";
             magic++;
             next;
         }
 
+        /^\. "`echo [^`]*`[^"]*"$/ {
+            tail = substr ($0, match ($0, /[^"]`/) +2);
+	    file = substr (tail, 1, length (tail) -1);
+	    system (sprintf ("env _RECURSE_MAGIC=%d '\'"$_G_qprogpath"\'' '\'"$_G_scriptdir"'%s'\''", magic, file));
+            next;
+        }
+
```

Comment:
To be honest, to review this - I need to have a deeper look and I run out of time ATM, so I'll have a look later if Gary isn't quicker.